### PR TITLE
feat: Use multipart formdata to start a job

### PR DESCRIPTION
**Pull request by arumsey**
- refactor to use `fetch` API
- convert JSON request to FormData
- add testing framework
- add bundler test

### CODE REVIEW

File: src/cmd/import.js [hunk 1]
```diff
@@ -46,12 +46,7 @@ export function importCommand(yargs) {
         stage
       } = argv;
 
-      // Ensure required arguments are provided and environment variables are set
-      if (!urlsPath) {
-        console.error(chalk.red('Usage: node import-helper.js --urls <path/to/urls.txt> [--options <options>] [--importjs <path/to/import.js>] [--stage]'));
```
Comment (arumsey) [inline src/cmd/import.js:51]: yargs takes care of this error now

File: src/cmd/import.js [hunk 2]
```diff
@@ -46,12 +46,7 @@ export function importCommand(yargs) {
...
-        process.exit(1);
-      }
-
+      // Ensure environment variables is set
```
Comment (blefebvre) [inline src/cmd/import.js]:
```suggestion
      // Ensure required environment variable is set
```

File: src/import/bundler.js [hunk 3]
```diff
@@ -10,18 +10,14 @@
...
 import esbuild from 'esbuild';
-
-function base64Encode(str) {
-  return Buffer.from(str).toString('base64');
-}
+import fs from 'fs';
 
 /**
- * Prepares the given import script by bundling it up and Base64 encoding it.
+ * Prepares the given import script by bundling it up and saving to the filesystem.
```
Comment (blefebvre) [inline src/import/bundler.js]: Does this function actually save the file or just return the text content?

File: test/import/bundler.test.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    prepareImportScript = await esmock('../../src/import/bundler.js', {
+      esbuild: { buildSync: buildSyncStub }
+    });
+  });
+
+  afterEach(() => {
+    sinon.restore();
+  });
+
+  it('should return a bundled the import script', () => {
```
Comment (blefebvre) [inline test/import/bundler.test.js]:
```suggestion
  it('should return a bundled import script', () => {
```

**Review by blefebvre — APPROVED** [review]
Thanks for introducing the test infrastructure! Few minor items.

File: src/import/bundler.js (hunk 3 above)
Comment (arumsey) [reply src/import/bundler.js]: At one point I had it saving the file but then moved to using Blob instead. The comment needs updating.

**Pull request merged**